compiler/src/iree/compiler/ConstEval/JitGlobals.cpp

```diff
@@ -448,6 +452,54 @@ static LogicalResult cloneUsedObjects(FunctionOpInterface funcOp,
   return success();
 }
 
+struct StripFlowTensorTransferPattern
+    : public OpRewritePattern<IREE::Flow::TensorTransferOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(IREE::Flow::TensorTransferOp op,
+                                PatternRewriter &rewriter) const override {
+    rewriter.replaceAllUsesWith(op.getResult(), op.getOperand());
+    rewriter.eraseOp(op);
+    return success();
+  }
+};
+
+// If an op that implements AffinityOpInterface has an optional stream affinity
+// attribute, remove it.
+struct StripStreamAffinityOptionalAttributePattern
+    : public OpInterfaceRewritePattern<IREE::Stream::AffinityOpInterface> {
+  using OpInterfaceRewritePattern::OpInterfaceRewritePattern;
+
+  LogicalResult matchAndRewrite(IREE::Stream::AffinityOpInterface op,
+                                PatternRewriter &rewriter) const override {
+    // Shouldn't we reject ops for which `op.requiresAffinity() == true`?
```

Comment:
if you're confused when you're writing this then someone coming across this in the future will be doubly confused

Reply:
I removed the comment. I was not sure if removing this attribute when it is required may result in an illegal operation.

Reply:
I guess since we are attaching the interface to Flow ops having the attribute missing must not lead to illegality.